### PR TITLE
Support addition of routes when handler is wrapped by functools.partial()

### PR DESCRIPTION
**Pull request by yuvalof**
## What do these changes do?
In aiohttp v4 a new check was added to make sure that route handlers are coroutines.
Python < v3.8 will return False for asyncio.iscoroutine(functools.partial(async_function)).
The change will iterate the wrapped function to make sure its async

## Are there changes in behavior for the user?
No

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/web_urldispatcher.py [hunk 1]
```diff
@@ -148,6 +149,12 @@ def __init__(self, method: str,
         self._expect_handler = expect_handler
         self._resource = resource
 
+    @staticmethod
+    def iscoroutinefunction(handler: Any) -> bool:
```
Comment (asvetlov) [inline aiohttp/web_urldispatcher.py]: The method should not be a part of public API.
Please make it first-class function which name is started from underscore.

Comment (yuvalof) [reply aiohttp/web_urldispatcher.py]: I exprted this function to helpers.py and changed it to only happen on Python<3.8

**Comment by asvetlov** [comment]
pls fix mypy: https://dev.azure.com/aio-libs/aiohttp/_build/results?buildId=1311&view=logs&j=cc4e4887-e527-5c83-25ce-76663593f2e1&t=9d25c5b1-1b8c-5097-b25c-a7f376352755&l=215

**Comment by asvetlov** [comment]
thanks!

**Comment by nicktimko** [comment]
Also fixes #3252

**Comment by asvetlov** [comment]
Thanks @nicktimko

**Pull request merged**